### PR TITLE
Integrate QuizPage with the app

### PR DESCRIPTION
**Pull request by SriramPatibanda**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your default branch!
- [x] Make sure you are making a pull request against the **default branch** (left side). Also you should start *your branch* off *default branch*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.
- [x] I have added necessary documentation (if appropriate)

### Description
Integrate QuizPage with the app using the GraphQL APIs.

❤️Thank you!

### Post merge checklist
- [ ] Follow steps from the [guidelines for contributing](../CONTRIBUTING.md) to this repository.
- [ ] If you are a new contributor, ping in the thread and one of the maintainers will add you to all-contributors list.

### Build Status
[pr-build-test](https://github.com/dscnitrourkela/project-maple/actions/workflows/pull_request.yml)

### CODE REVIEW

File: aptiche/lib/datamodels/api_models.dart [hunk 1]
```diff
@@ -2,22 +2,36 @@ class Question {
   Question({
     required this.questionId,
     required this.question,
-    required this.url,
+    // required this.url,
```
Comment (abhibhaw) [inline aptiche/lib/datamodels/api_models.dart:5]: Remove unnecessary comments 😆

**Review by abhibhaw — CHANGES_REQUESTED** [review]

File: aptiche/lib/datamodels/api_models.dart (hunk 1 above)
Comment (abhibhaw) [reply aptiche/lib/datamodels/api_models.dart:5]: PS: I am still reviewing

File: aptiche/lib/views/dataentry/dataentry.dart [hunk 2]
```diff
@@ -264,6 +264,8 @@ class DataEntryScreen extends GetView<DataEntryController> {
                           width: SizeConfig.screenWidth,
                           alignment: Alignment.center,
                           child: CustomButton(
+                              horizontalPadding: 15,
```
Comment (nayakastha) [inline aptiche/lib/views/dataentry/dataentry.dart:267]: Use relative values. SizeConfig

File: aptiche/lib/views/home/home_controller.dart [hunk 3]
```diff
@@ -4,45 +4,60 @@ import 'package:aptiche/utils/enums.dart';
...
+    pastQuizzes = await _graphQL.getQuizzesByTime(1);
+    activeQuizzes = await _graphQL.getQuizzesByTime(2);
+    upcomingQuizzes = await _graphQL.getQuizzesByTime(3);
+    desiredList = upcomingQuizzes;
+    homeState.value = CurrentState.ready;
     update();
   }
 
   @override
+  // ignore: unnecessary_overrides
```
Comment (nayakastha) [inline aptiche/lib/views/home/home_controller.dart:60]: Why was this override needed? is onClose( ) compulsory?

File: aptiche/lib/views/login/loginscreen.dart [hunk 4]
```diff
@@ -117,6 +117,8 @@ class LoginView extends GetView<LoginController> {
                               )
                             else
                               CustomButton(
+                                horizontalPadding: 15,
```
Comment (nayakastha) [inline aptiche/lib/views/login/loginscreen.dart:120]: Same thing here as well. SizeConfig pls

**Review by nayakastha — CHANGES_REQUESTED** [review]
Haven't tried running the app, but the code seems to work fine logically.
Just take care not to use absolute values for the sizes, padding and magins

File: aptiche/lib/utils/enums.dart [hunk 5]
```diff
@@ -1 +1,3 @@
 enum CurrentState { busy, ready }
+
+enum ChoicesEnum { optionA, optionB, optionC, optionD, NON }
```
[the remark this answers is not in the document]
Comment (SriramPatibanda) [reply aptiche/lib/utils/enums.dart:3]: I think so, not really sure

File: aptiche/lib/views/home/home_controller.dart (hunk 3 above)
Comment (SriramPatibanda) [reply aptiche/lib/views/home/home_controller.dart:60]: Yeah, it is kinda.

**Review by nayakastha — APPROVED** [review]

**Review by riteshsp2000 — APPROVED** [review]

**Pull request merged**